### DISCUSSION TITLE
mtx-slider thumblabel format & initial visibility

### DISCUSSION THREAD

**Discussion by crisconz** [ng-matero/extensions#76] [Ideas]
Will you be planning on supporting?

1) A customisable thumb label - like in material = https://material.angular.io/components/slider/overview#formatting-the-thumb-label

2) To allow the thumb label to be visible by default without having to click or move the slider

**Comment by crisconz** [comment]
Update - for:
 
1) I see the [displayWith] property nearly works.
However the first time you click on the slider - the custom text only updates either the left or the right slider thumbnail (pending which slider gets the event). The other says undefined.
![image](https://user-images.githubusercontent.com/94721222/142708059-fec81a8c-d9ac-4302-b0ea-ea55ba122ae8.png)

2) We can modify the answer given in the below link. We simply need to change mat to mtx in the css - eg  .mtx-slider-thumb vs .mat-slider-thumb 
 https://stackoverflow.com/questions/58057900/how-can-i-display-thumb-label-all-the-time-for-material-slider

Applying 1 & 2 - I now see undefined in both my thumbnails (left & right)
![image](https://user-images.githubusercontent.com/94721222/142708889-76b0061c-62ee-489e-8030-d033b38dd053.png)

Still need to work out how to fix that...

**Reply by crisconz** [reply]
Ahh...

Turns out that I had initialised the slider incorrectly by put an invalid number into my value range.  Basic mistake. But that explains why the thumbnails didn't show correctly on initial load (but they did after a click set a good value).
All good now. 
![image](https://user-images.githubusercontent.com/94721222/142709629-a5ba86bc-e43f-488c-af17-a41dc61831cb.png)
 
So this thread may only be useful for someone for point #2. -> making the thumb label to be visible by default
Otherwise ignore it!

**Comment by nzbin** [comment]
Many thanks your advice, I am busy with work now. I know the range slider has many defects, I will improve it in the near future.
